### PR TITLE
Add js_of_ocaml support.

### PR DESCRIPTION
**Pull request by mefyl**

### CODE REVIEW

File: src/clock.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  // First 24 bits.
+  var low = us & 0xffffff;
+  // Mid 24 bits.
+  var mid = Math.trunc(us / Math.pow(2, 24)) & 0xffffff;
+  // Upper 16 bits.
+  var hi = Math.trunc(us / Math.pow(2, 48)) & 0xffff;
+
+  // Using MlInt64 is necessary for the return value to interact correctly with
+  // OCaml code.
+  return new MlInt64(low, mid, hi);
```
Comment (nojb) [inline src/clock.js:29]: What about using `caml_int64_of_float` instead?
```suggestion
  return caml_int64_of_float(rawClockMs);
```

Comment (mefyl) [reply src/clock.js]: Works just as well indeed 👍

File: src/clock.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+//Provides: caml_highres_clock const
+//Requires: MlInt64
+
```
Comment (nojb) [inline src/clock.js:3]:
```suggestion
//Requires: caml_int64_of_float
```

File: src/clock.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  // - 1ms precision on Firefox [1]
+  // - 5us to 100us precision on Chromium [2]
+  //
+  // [1]: https://developer.mozilla.org/en-US/docs/Web/API/Performance/now#reduced_time_precision
+  // [2]: https://chromestatus.com/feature/6497206758539264
+  //
+  // We could probably get vastly better accuracy under Electron by using a Node.js runtime.
+  var rawClockMs = globalThis.performance.now();
+
+  return caml_int64_of_float(rawClockMs);
```
Comment (nojb) [inline src/clock.js:17]: Nit:
```suggestion
  return caml_int64_of_float(globalThis.performance.now());
```

**Review by nojb — COMMENTED** [review]
I think the `Requires` annotation needs fixing.

**Review by nojb — APPROVED** [review]
LGTM. Can you add a `CHANGES.md` entry? Thanks!

**Comment by nojb** [comment]
Thanks!

**Pull request merged**